### PR TITLE
sys-auth/authenticator: version bump

### PR DESCRIPTION
**Pull request by linxon**
Closes: https://bugs.gentoo.org/696476
Signed-off-by: Yury Martynov <email@linxon.ru>

### CODE REVIEW

File: dev-python/text-unidecode/text-unidecode-1.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="The most basic Text::Unidecode port"
+HOMEPAGE="https://github.com/kmike/text-unidecode/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
```
Comment (linxon) [inline dev-python/text-unidecode/text-unidecode-1.3.ebuild]: I'm not sure about this file https://github.com/kmike/text-unidecode/blob/a0005d7a0b43a6aa21ce8a63f1da48c6c44f03e1/src/text_unidecode/data.bin

but I still hope it using for decoding unicode chars :)
https://github.com/kmike/text-unidecode/blob/a0005d7a0b43a6aa21ce8a63f1da48c6c44f03e1/src/text_unidecode/__init__.py#L6-L21

File: sys-auth/authenticator/authenticator-3.32.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="gnome-shell test"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RESTRICT="!test? ( test )"
+
+CDEPEND="${PYTHON_DEPS}
```
Comment (juippis) [inline sys-auth/authenticator/authenticator-3.32.2.ebuild:31]: Use `COMMON_DEPEND` instead.

File: sys-auth/authenticator/authenticator-3.32.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	app-crypt/libsecret[introspection]
+	>=gui-libs/libhandy-0.0.11:=[introspection]
```
Comment (juippis) [inline sys-auth/authenticator/authenticator-3.32.2.ebuild:33]: This doesn't seem to have a sublost specified, nor are there more than 1 version available in ::gentoo. `gui-libs/libhandy`

File: sys-auth/authenticator/authenticator-3.32.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	x11-libs/gtk+:3[introspection]
+	$(python_gen_cond_dep '
+		dev-python/beautifulsoup:4[${PYTHON_MULTI_USEDEP}]
+		dev-python/pillow[${PYTHON_MULTI_USEDEP}]
+		dev-python/pyfavicon[${PYTHON_MULTI_USEDEP}]
+		dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]
+		dev-python/pyotp[${PYTHON_MULTI_USEDEP}]
+		dev-python/pyzbar[${PYTHON_MULTI_USEDEP}]
+		dev-python/yoyo-migrations[${PYTHON_MULTI_USEDEP}]
```
Comment (juippis) [inline sys-auth/authenticator/authenticator-3.32.2.ebuild:42]: I think `PYTHON_USEDEP` works here.

File: sys-auth/authenticator/authenticator-3.32.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+else
+	SRC_URI="https://gitlab.gnome.org/World/Authenticator/-/archive/${PV}/Authenticator-${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+
+	S="${WORKDIR}/Authenticator-${PV}"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="gnome-shell test"
```
Comment (juippis) [inline sys-auth/authenticator/authenticator-3.32.2.ebuild:26]: `gnome-shell` USE doesn't seem to be having any function in this ebuild, then it shouldn't be used also.

File: dev-python/yoyo-migrations/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>oleksandr.trotsenko@gmail.com</email>
+		<name>Oleksandr Trotsenko</name>
+	</maintainer>
```
Comment (juippis) [inline dev-python/yoyo-migrations/metadata.xml:7]: Take him out.

File: dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="A database schema migration tool"
+HOMEPAGE="https://ollycope.com/software/yoyo"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
```
Comment (juippis) [inline dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild:16]: You can't add `arm` or `arm64` keywords, they need to be requested through bugzilla.

File: dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="A database schema migration tool"
+HOMEPAGE="https://ollycope.com/software/yoyo"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
```
Comment (juippis) [inline dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild:12]: I see they have at least one testfile included. Could you try implementing tests in this ebuild?

File: dev-python/text-unidecode/text-unidecode-1.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="The most basic Text::Unidecode port"
+HOMEPAGE="https://github.com/kmike/text-unidecode/"
```
Comment (juippis) [inline dev-python/text-unidecode/text-unidecode-1.3.ebuild:11]: Add Github here as 2ndary homepage. 

Add `python_test`.

File: dev-python/text-unidecode/text-unidecode-1.3.ebuild [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
```
Comment (juippis) [inline dev-python/text-unidecode/text-unidecode-1.3.ebuild:16]: You can't add `arm` or `arm64` here.

File: dev-python/text-unidecode/metadata.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>oleksandr.trotsenko@gmail.com</email>
+		<name>Oleksandr Trotsenko</name>
+	</maintainer>
```
Comment (juippis) [inline dev-python/text-unidecode/metadata.xml:7]: Take him out.

File: dev-python/pyfavicon/pyfavicon-0.1.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Async favicon fetcher"
+HOMEPAGE="https://github.com/bilelmoussaoui/pyfavicon"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
```
Comment (juippis) [inline dev-python/pyfavicon/pyfavicon-0.1.1.ebuild:16]: Take extra keywords out. Add python tests to this ebuild.

File: dev-python/pyfavicon/metadata.xml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>oleksandr.trotsenko@gmail.com</email>
+		<name>Oleksandr Trotsenko</name>
+	</maintainer>
```
Comment (juippis) [inline dev-python/pyfavicon/metadata.xml:7]: Take him out.

File: dev-python/iniherit/metadata.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>oleksandr.trotsenko@gmail.com</email>
+		<name>Oleksandr Trotsenko</name>
+	</maintainer>
```
Comment (juippis) [inline dev-python/iniherit/metadata.xml:7]: Here too.

File: dev-python/iniherit/iniherit-0.3.9.ebuild [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="A ConfigParser subclass with file-specified inheritance"
+HOMEPAGE="https://github.com/cadithealth/iniherit"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
```
Comment (juippis) [inline dev-python/iniherit/iniherit-0.3.9.ebuild:16]: And here.

**Comment by okias** [comment]
@linxon hey, do you plan finishing it? If yes, i'll do review & test.

File: sys-auth/authenticator/authenticator-3.32.2.ebuild (hunk 5 above)
Comment (linxon) [reply sys-auth/authenticator/authenticator-3.32.2.ebuild:26]: it reqs `gnome-base/gnome-shell` for QR code scanning
```bash
[ERROR] 2020-05-15 13:07:27 Could not read the secret code,the keyring keys were reset manually
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/Authenticator/widgets/accounts/add.py", line 236, in scan_qr
    filename = GNOMEScreenshot.area()
  File "/usr/lib64/python3.6/site-packages/Authenticator/models/screenshot.py", line 56, in area
    -1, None).unpack()
gi.repository.GLib.Error: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.Shell.Screenshot was not provided by any .service files (2)
```

Selection_102

File: sys-auth/authenticator/authenticator-3.32.2.ebuild (hunk 4 above)
Comment (linxon) [reply sys-auth/authenticator/authenticator-3.32.2.ebuild:42]: using `PYTHON_USEDEP` I get error: 
```bash
  dependency.syntax [fatal]     4
   sys-auth/authenticator/authenticator-9999.ebuild: DEPEND: Invalid atom (Invalid use dep: '%PYTHON_USEDEP-NEEDS-TO-BE-USED-IN-PYTHON_GEN_COND_DEP%'), token 19
```

File: dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild (hunk 8 above)
Comment (linxon) [reply dev-python/yoyo-migrations/yoyo-migrations-7.0.0.ebuild:12]: unittest:
```
Ran 0 tests in 0.000s
```

pytest
```
no tests ran in 0.11 seconds
```

**Comment by SoapGentoo** [comment]
This package is on its way out. If you want to save it, please rebase and freshen up the ebuilds. Also, add python 3.9 support everywhere.

**Pull request closed without merging**